server/src/main/java/org/opensearch/cluster/node/DiscoveryNodeRole.java

```diff
@@ -290,11 +290,33 @@ public Setting<Boolean> legacySetting() {
 
     };
 
+    /**
+     * Represents the role for a search node, which is dedicated to provide search capability.
+     */
+    public static final DiscoveryNodeRole SEARCH_ROLE = new DiscoveryNodeRole("search", "s", true) {
+
+        @Override
+        public Setting<Boolean> legacySetting() {
+            // search role is added in 2.4 so doesn't need to configure legacy setting
+            return null;
+        }
+
+        @Override
+        public DiscoveryNodeRole getCompatibilityRole(Version nodeVersion) {
+            if (nodeVersion.onOrAfter(Version.V_2_4_0)) {
+                return this;
+            } else {
+                return DiscoveryNodeRole.DATA_ROLE;
```

Comment:
Do we necessarily need a fallback?

Reply:
Let me check what will be like when there is no fallback. So far I think without the fallback, the new role will be taken as "UnknownRole" when communicating with node in older versions, but I need to figure out the impact.
https://github.com/opensearch-project/OpenSearch/blob/00fb2b72dc253c5697e5bc2410830b66879d6bc3/server/src/main/java/org/opensearch/cluster/node/DiscoveryNodeRole.java#L314